peachjam/middleware.py

```diff
@@ -48,6 +49,14 @@ def process_request(self, request):
             del request.META["HTTP_ACCEPT_LANGUAGE"]
 
 
+class SetPreferredLanguageMiddleware(MiddlewareMixin):
+    def process_response(self, request, response):
+        if hasattr(request, "set_language"):
+            request.POST = {"language": request.set_language}
+            response = set_language(request)
+        return response
```

Comment:
@longhotsummer i've added the middleware here to set the language when logging in. we make use of the set_language view. it does a redirect and checks the next parameter. this is better than setting the cookie manually

Reply:
this redirect is breaking some tests :(